airflow/operators/python.py

```diff
@@ -248,7 +248,7 @@ def execute(self, context: Dict):
 
         if condition:
             self.log.info('Proceeding with downstream tasks...')
-            return
+            return condition
```

Comment:
The value of condition is always True in this case.

Reply:
I agree. But it might be some non-empty string or list

Reply:
How can it be a list? the condition is boolean value.
Am I missing something here? This is ShortCircut not PythonBranch.

Reply:
I can't see where defined that ShortCircuitOperator's callable should return only boolean.
Maybe I'm missing something.
Here is an example of what I mean. In the first case, callable returns a non-empty string, and ShortCircuitOperator passes.
In the second case return value is the empty list, which is casts as False.

```
from datetime import datetime
from airflow.models import DAG
from airflow.operators.python import ShortCircuitOperator
from airflow.operators.dummy import DummyOperator

with DAG('test_short_circuit_dag', start_date=datetime(2021, 12, 7)) as dag:
    circuit_true = ShortCircuitOperator(task_id='circuit_true', python_callable=lambda: 'signature')
    check_true = DummyOperator(task_id='check_true')
    circuit_true >> check_true

    circuit_false = ShortCircuitOperator(task_id='circuit_false', python_callable=lambda: [])
    check_false = DummyOperator(task_id='check_false')
    circuit_false >> check_false
```

<img width="303" alt="Screenshot 2021-12-08 at 18 33 19" src="https://user-images.githubusercontent.com/10523218/145247230-f0692172-e847-4ea2-a102-6f0ce550fba4.png">

Reply:
@eladkal `condition` result is not necessarily a boolean - it can be but in a Boolean operation (like `if`), other variables evaluates to True/False - see https://docs.python.org/3/reference/expressions.html#booleans
Some refer to that evaluation as Falsey/Truthy.
If you feel returning a boolean is useless and a burden, we could do
`return condition if isinstance(condition, boolean) else None`

Reply:
@yiga2 So just to clarify the feature request was to push to Xcom whatever value the python_callable returns?

Reply:
e.g. as per @kazanzhy example, an empty list evaluates to `False` while a non-empty evaluates to `True` and that list can be used by tasks downstreams. This applies to list() and other objects - see many examples online.